Client-Observability-Java/app/src/main/java/com/tokbox/sample/clientobservability/MainActivity.java

```diff
@@ -0,0 +1,283 @@
+
+            publisher = new Publisher.Builder(MainActivity.this).senderStatsTrack(true).build();
+            publisher.setPublisherListener(publisherListener);
+            publisher.getRenderer().setStyle(BaseVideoRenderer.STYLE_VIDEO_SCALE, BaseVideoRenderer.STYLE_VIDEO_FILL);
+            
+            publisherViewContainer.addView(publisher.getView());
+
+            if (publisher.getView() instanceof GLSurfaceView) {
+                ((GLSurfaceView) publisher.getView()).setZOrderOnTop(true);
+            }
+
+            session.publish(publisher);
+        }
+
+        @Override
+        public void onDisconnected(Session session) {
+            Log.d(TAG, "onDisconnected: Disconnected from session: " + session.getSessionId());
+        }
+
+        @Override
+        public void onStreamReceived(Session session, Stream stream) {
+            Log.d(TAG, "onStreamReceived: New Stream Received " + stream.getStreamId() + " in session: " + session.getSessionId());
+
+            if (subscriber == null) {
+                subscriber = new Subscriber.Builder(MainActivity.this, stream).build();
+                subscriber.getRenderer().setStyle(BaseVideoRenderer.STYLE_VIDEO_SCALE, BaseVideoRenderer.STYLE_VIDEO_FILL);
+                subscriber.setSubscriberListener(subscriberListener);
+                subscriber.setVideoStatsListener(videoStatsListener);
+                session.subscribe(subscriber);
+                subscriberViewContainer.addView(subscriber.getView());
+            }
+        }
+
+        @Override
+        public void onStreamDropped(Session session, Stream stream) {
+            Log.d(TAG, "onStreamDropped: Stream Dropped: " + stream.getStreamId() + " in session: " + session.getSessionId());
+
+            if (subscriber != null) {
+                subscriber = null;
+                subscriberViewContainer.removeAllViews();
+            }
+        }
+
+        @Override
+        public void onError(Session session, OpentokError opentokError) {
+            finishWithMessage("Session error: " + opentokError.getMessage());
+        }
+    };
+
+    SubscriberKit.SubscriberListener subscriberListener = new SubscriberKit.SubscriberListener() {
+        @Override
+        public void onConnected(SubscriberKit subscriberKit) {
+            Log.d(TAG, "onConnected: Subscriber connected. Stream: " + subscriberKit.getStream().getStreamId());
+        }
+
+        @Override
+        public void onDisconnected(SubscriberKit subscriberKit) {
+            Log.d(TAG, "onDisconnected: Subscriber disconnected. Stream: " + subscriberKit.getStream().getStreamId());
+        }
+
+        @Override
+        public void onError(SubscriberKit subscriberKit, OpentokError opentokError) {
+            finishWithMessage("SubscriberKit onError: " + opentokError.getMessage());
+        }
+    };
+
+    SubscriberKit.VideoStatsListener videoStatsListener = new SubscriberKit.VideoStatsListener() {
```

Comment:
I understand it only shows subscriber side stats because we want to focus on sender-side stats. Maybe we can extend with all the new (mainly publisher) ones after the 2.33 release.